articles/scripting-extensions.md

```diff
@@ -0,0 +1,94 @@
+---
+uid: scripting-extensions
+title: "Scripting Extensions"
+---
+
+# Scripting Extensions
+
+Local extensions can be used to augment the functionality of workflows with new operators scripted directly using the C# programming language. This gives you the ability to tap into the entire .NET ecosystem without requiring a more formal C# project, as the Bonsai bootstrapper will automatically compile the scripts from source each time the workflow is loaded.
+
+Scripting extensions are more commonly used to develop small project-specific functionality, but can also be easily shared with others by simply copying individual scripts.
+
+> [!Tip]
+> If you are interested in developing extensions to be reused across several different projects, consider [**creating a package**](xref:create-package) instead.
+
+## Pre-requisites
+
+1. [.NET 6.0 SDK](https://dotnet.microsoft.com/en-us/download).
+2. [Visual Studio Code](https://code.visualstudio.com/).
+3. [C# extension for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csharp).
+4. [.NET Framework 4.7.2 Developer Pack](https://dotnet.microsoft.com/en-us/download/dotnet-framework/net472).
+
+> [!Warning]
+> You might need to reboot your machine after installing these components.
+
+## Creating a local extension
+
+1. From the Bonsai editor, either open an existing workflow, or create and save a new workflow.
+
+2. Add a new `CSharpSource` operator from the **Toolbox**.
+
+3. Double-click on the scripting node. A dialog should be displayed asking whether you want to create the **Extensions** folder. Click `Yes` to proceed and give a name to your new scripting operator. If the above [pre-requisites](#pre-requisites) have been installed, Visual Studio Code should then be launched automatically.
+
+4. After Visual Studio Code opens, a notification should appear on the lower-right corner with a warning about unresolved dependencies. Click the `Restore` button to finish configuring the extensions project. You can now go ahead and modify your local extension script.
+
+    ![Restore unresolved dependencies](~/images/scripting-restoredependencies.png)
+
+    > [!Note]
+    > If the notification disappears before you have a chance to click `Restore`, you can bring back the notification by clicking the small blue bell icon on the lower-right corner of Visual Studio Code.
+
+5. Every Bonsai operator specifies an observable sequence using the <xref href="System.IObservable`1"/> interface. The [System.Reactive](http://reactivex.io/) package provides a comprehensive library of methods used to generate and manipulate observable sequences in C#. The simplest way to implement new operators is by using the methods in the <xref href="System.Reactive.Linq.Observable"/> class.
```

Comment:
Consider adding Transform example as those tend to be generally more used and it is helpful to explore the syntax of inputs and outputs to the method;
Also, maybe adding an example of how to create a struct might be interesting!

Reply:
Yeah, these are both good points. The old example was a Transform, but in the new project template the default operator that is placed in the project is a source. However, maybe we can also make a new transform operator to make something interesting on the source.